libsolidity/codegen/LValue.cpp

```diff
@@ -300,9 +300,12 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 			// stack: value storage_ref cleared_value multiplier
 			utils.copyToStackTop(3 + m_dataType->sizeOnStack(), m_dataType->sizeOnStack());
 			// stack: value storage_ref cleared_value multiplier value
-			if (FunctionType const* fun = dynamic_cast<decltype(fun)>(m_dataType))
+			if (auto const* fun = dynamic_cast<FunctionType const*>(m_dataType))
 			{
-				solAssert(_sourceType == *m_dataType, "function item stored but target is not equal to source");
+				solAssert(
```

Comment:
Can you add an assertion that the function types are both not bound and that their stack sizes are 1?

Reply:
Actually let me just quickly do it...